great_expectations/expectations/core/expect_column_values_to_be_of_type.py

```diff
@@ -362,6 +366,22 @@ def _validate_sqlalchemy(self, actual_column_type, expected_type, execution_engi
                         type_module, expected_type
                     )
                     types.append(potential_type)
+                elif type_module.__name__ == "snowflake.sqlalchemy.snowdialect":
+                    normalized_sfsqlalchemy_type_name = (
+                        normalize_snowflake_data_type_name(actual_column_type)
+                    )
+                    normalized_expected_type_name = normalize_snowflake_data_type_name(
+                        expected_type
+                    )
+                    normalized_native_type_name = map_sfsqlalchemy_type_to_sf_type(
+                        normalized_sfsqlalchemy_type_name
+                    )
+                    return {
+                        "success": normalized_native_type_name
+                        == normalized_expected_type_name,
+                        "result": {"observed_value": normalized_native_type_name},
+                    }
```

Comment:
It is fair to think that changing the metric would be a breaking change for other Expectations. Especially since (for SQL) we are using the column metadata, and that metric is more widely used.